packages/core/src/http-client/http-request-config.ts

```diff
@@ -0,0 +1,40 @@
+import { createLogger, pickWithoutKeys } from '@sap-cloud-sdk/util';
+
+const logger = createLogger({
+  package: 'core',
+  messageContext: 'http-request-config'
+});
+
+/**
+ * Filter disallowed keys from a given custom request configs object.
+ * @param customRequestConfigs a given custom request configs object to be filtered
+ * @param disallowedKeys A list of keys that are not allowed to be customized.
+ * @returns Filtered custom request configs object.
+ */
+export function filterCustomRequestConfigs(
```

Comment:
I think this is one configuration with multiple configuration options, not multiple configurations, therefore I would like to refrain from using the word "configs".
Please cross check everywhere (also docs).